### PR TITLE
dump boskos metrics for all resource types

### PR DESCRIPTION
**Pull request by krzyzacy**
this looks better?

xref https://github.com/kubernetes/test-infra/issues/7771

/assign @BenTheElder

### CODE REVIEW

**Comment by krzyzacy** [comment]
I'll take https://github.com/kubernetes/test-infra/pull/8214 in first

File: boskos/metrics/metrics.go [hunk 1]
```diff
@@ -30,62 +32,57 @@ import (
...
-				Help: "Number of cleaning gce-project",
-			}),
-		},
-		GkeStats: map[string]prometheus.Gauge{
+		BoskosState: map[string]map[string]prometheus.Gauge{},
+	}
+
+	// I need to access to the list of projects in init, so I cannot do a flag :-\
+	resources = []string{
+		"gce-project",
```
Comment (sebastienvas) [inline boskos/metrics/metrics.go]: It should not be hard coded! This is not usable for istio.

Comment (krzyzacy) [reply boskos/metrics/metrics.go]: ok, I guess I'll look into a way to access flags in init? or I can put that into an env var?

Comment (krzyzacy) [reply boskos/metrics/metrics.go]: fixed

File: boskos/metrics/metrics.go [hunk 2]
```diff
@@ -30,58 +32,46 @@ import (
...
+		BoskosState: map[string]map[string]prometheus.Gauge{},
+	}
+	resources common.ResTypes
+)
+
+func init() {
+	flag.Var(&resources, "resource-type", "comma-separated list of resources need to have metrics collected")
+}
+
+func init_prometheus() {
```
Comment (BenTheElder) [inline boskos/metrics/metrics.go]: nit: this is not a very go like signature

Comment (krzyzacy) [reply boskos/metrics/metrics.go]: sure

**Comment by krzyzacy** [comment]
PTAL :-)

**Pull request merged**